Automattic/jetpack
# Payment blocks: Add donation and allow custom amount options

Added two new configuration options for marking a membership subscription as a donation and an option that allows customers to pay a custom amount (whatever they want).





Fixes https://github.com/Automattic/jetpack/issues/23416
Fixes https://github.com/Automattic/jetpack/issues/23415

#### Changes proposed in this Pull Request:

* Added two toggles in the Product management system of the Payment blocks (Premium content and Payment button).

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Apply `arc patch D79089 && arc patch D77465 --nobranch && arc patch D77864 --nobranch` on your WPCOM sandbox. 
* Atomic: You'll need to use your WoA dev site and merge locally the branch from https://github.com/Automattic/jetpack/pull/23714 and sync your WoA site with these changes
* Atomic: add `define( 'JETPACK__SANDBOX_DOMAIN', 'your-sandbox');` to your wp-config.php
* Go to wp-admin/post-new.php and add a new payment block (Premium content or Payment button).
* Add a new subscription.
* Make sure that "Mark as a donation" and "Enable customers to pick their own amount" toggles are working properly.
* Use browser's DevTools and make sure that the POSt /product request contains the "type", "buyer_can_change_amount" and "public" properties as payload.
* Use [Calypso Live link](https://github.com/Automattic/wp-calypso/pull/62208#issuecomment-1077489189) and go to /earn/payment-plans and make sure the newly created subscription has the options enabled.
